istioctl/cmd/vm-bootstrap.go

```diff
@@ -0,0 +1,612 @@
+}
+
+func copyCertificates(kubeClient kubernetes.Interface,
+	addressCertMapping map[string]workloadEntryAddressKeys) error {
+	var callback ssh.HostKeyCallback
+	if sshIgnoreHostKeys {
+		callback = ssh.InsecureIgnoreHostKey()
+	} else {
+		user, err := user.Current()
+		if err != nil {
+			return err
+		}
+		callback, err = knownhosts.New(path.Join(user.HomeDir, ".ssh", "known_hosts"))
+		if err != nil {
+			return err
+		}
+	}
+
+	for address, certs := range addressCertMapping {
+		sshConfig := &ssh.ClientConfig{
+			User:            sshUser,
+			Auth:            []ssh.AuthMethod{sshAuthMethod},
+			HostKeyCallback: callback,
+		}
+		client, err := ssh.Dial("tcp", address+":22", sshConfig)
+		if err != nil {
+			return err
+		}
+		// Ensure the remote directory exists.
+		mkdirErr := remoteRunCommand("mkdir -p "+remoteDirectory, client)
+		if mkdirErr != nil {
+			client.Close()
+			return err
+		}
+
+		remoteCaPath := path.Join(remoteDirectory, "ca-cert-"+address+".pem")
+		err = remoteCopyFile(certs.CaCert, remoteCaPath, client)
+		if err != nil {
+			client.Close()
+			return err
+		}
+
+		remoteCertPath := path.Join(remoteDirectory, "cert-"+address+".pem")
+		err = remoteCopyFile(certs.Cert, remoteCertPath, client)
+		if err != nil {
+			client.Close()
+			return err
+		}
+		remoteKeyPath := path.Join(remoteDirectory, "key-"+address+".pem")
+		err = remoteCopyFile(certs.Key, remoteKeyPath, client)
+		if err != nil {
+			client.Close()
+			return err
+		}
+
+		if startIstio {
+			addressIdentifier := addressToPodNameAddition(address)
+			istiodPort, err := deriveDiscoveryAddressPort(certs.ServiceName, certs.ServiceNamespace, kubeClient)
+			if err != nil {
+				client.Close()
+				return err
+			}
+			discoveryAddress := "istiod.istio-system.svc:" + istiodPort
+			var authPolicyStr string
+			if mutualTLS {
+				authPolicyStr = "MUTUAL_TLS"
+			} else {
+				authPolicyStr = "NONE"
+			}
+
+			meshPath := path.Join(remoteDirectory, "mesh.yaml")
+			err = remoteCopyFile([]byte(`---
+defaultConfig:
+	serviceCluster:`+certs.ServiceNamespace+`
+	controlPlaneAuthPolicy: `+authPolicyStr+`
+	discoveryAddress: `+discoveryAddress), meshPath, client)
+			if err != nil {
+				client.Close()
+				return err
+			}
+
+			//you need to deal with Sidecar CR if you want it to be "non-captured" mode
+			err = remoteRunCommand(
+				"docker run -d --name istio-proxy --network host "+
+					"-v "+meshPath+":/etc/istio/config/mesh "+
+					"-v "+remoteCaPath+":/var/run/secrets/istio/root-cert.pem "+
+					"-v "+remoteCertPath+":/var/run/secrets/istio/cert-chain.pem "+
+					"-v "+remoteKeyPath+":/var/run/secrets/istio/key.pem "+
+					"--add-host istio-pilot.istio-system.svc:"+address+" "+
+					"--add-host istiod.istio-system.svc:"+address+" "+
+					"-e POD_NAME="+certs.ServiceName+"-"+addressIdentifier+" "+
+					"-e JWT_POLICY=none "+
+					"-e PROV_CERT=/var/run/secrets/istio "+
+					"-e PILOT_CERT_PROVIDER=istiod "+
```

Comment:
we will also need to set `-e OUTPUT_CERTS=var/run/secrets/istio` such that the key/cert will be rotated